apps/condo/domains/resident/schema/RegisterResidentService.js

```diff
@@ -30,14 +31,28 @@ const RegisterResidentService = new GQLCustomSchema('RegisterResidentService', {
                     unitName,
                     user: { connect: { id: context.authedItem.id } },
                 }
-                const [property] = await Property.getAll(context, { address })
-                if (property) {
-                    attrs.property = { connect: { id: property.id } }
-                    attrs.organization = { connect: { id: property.organization.id } }
+                const [existingResident] = await ResidentAPI.getAll(context, {
+                    address,
+                    unitName,
+                    user: { id: context.authedItem.id },
+                })
+                let id
+                if (existingResident) {
+                    await ResidentAPI.update(context, existingResident.id, {
+                        deletedAt: null,
+                    })
+                    id = existingResident.id
+                } else {
+                    const [property] = await Property.getAll(context, { address })
+                    if (property) {
+                        attrs.property = { connect: { id: property.id } }
+                        attrs.organization = { connect: { id: property.organization.id } }
+                    }
+                    const resident = await Resident.create(context, attrs)
+                    id = resident.id
                 }
-                const resident = await Resident.create(context, attrs)
                 // Hack that helps to resolve all subfields in result of this mutation
-                return await getById('Resident', resident.id)
+                return await getById('Resident', id)
```

Comment:
better not to use return await
[link](http://hassansin.github.io/Why-return-await-Is-a-Bad-Idea)

Reply:
Changed at `0a38781`.